src/ngx_http_lua_socket_tcp.c

```diff
@@ -4908,65 +4901,30 @@ ngx_http_lua_socket_push_input_data(ngx_http_request_t *r,
     nbufs = 0;
     ll = NULL;
 
+    luaL_buffinit(L, &luabuf);
     for (cl = u->bufs_in; cl; cl = cl->next) {
+        ptrdiff_t chunk_size;
```

Comment:
Agreed. This is according to the nginx coding style (though declaring the local in the inner-most scope using it is actually more efficient from the compiler's perspective).